pkg/kv/postgres/store.go

```diff
@@ -260,22 +271,32 @@ func (s *Store) Scan(ctx context.Context, partitionKey, start []byte) (kv.Entrie
 		return nil, kv.ErrMissingPartitionKey
 	}
 
+	return s.scanInternal(ctx, partitionKey, start, true)
+}
+
+func (s *Store) scanInternal(ctx context.Context, partitionKey, start []byte, includeStart bool) (*EntriesIterator, error) {
 	var (
 		rows pgx.Rows
 		err  error
 	)
 
 	if start == nil {
-		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 ORDER BY key`, partitionKey)
+		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 ORDER BY key LIMIT $2`, partitionKey, s.Params.ScanPageSize)
 	} else {
-		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 AND key >= $2 ORDER BY key`, partitionKey, start)
+		compareOp := ">="
+		if !includeStart {
+			compareOp = ">"
+		}
+		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 AND key `+compareOp+` $2 ORDER BY key LIMIT $3`, partitionKey, start, s.Params.ScanPageSize)
 	}
 
 	if err != nil {
 		return nil, fmt.Errorf("%s: %w (start=%v)", err, kv.ErrOperationFailed, start)
 	}
 	return &EntriesIterator{
-		rows: rows,
+		ctx:   ctx,
+		rows:  rows,
```

Comment:
I think the most important part is not fixed here. The connection to postgres is being held by the rows.
Look at `loadBuffer` of `DBIterator` to understand the differences.

Reply:
Rows are now scanned and "converted" to entries as part of `scanInternal`. This allows `rows.Close`, and the implied connection release, before iterations start.
Hop that what you meant.

Ran the same BM. Results are similar - again, it is a local PG instance, so no real network latency
```
time ./lakectl fs ls lakefs://load-test/main/abuse-for-list/ | wc -l 
100000

real	0m4.028s
user	0m1.633s
sys	0m1.046s
```